sa/database_test.go

```diff
@@ -73,3 +73,26 @@ func TestStrictness(t *testing.T) {
 		t.Fatalf("Got wrong type of error: %s", err)
 	}
 }
+
+func TestSafeSelectOne(t *testing.T) {
```

Comment:
Sounds good. I don't feel strongly. I don't think its worth carrying a lot more mock code if the other tests are exercising the same code paths.